packages/dds/tree/src/test/simple-tree/core/allowedTypes.spec.ts

```diff
@@ -567,6 +573,233 @@ describe("allowedTypes", () => {
 			type X6 = InsertableObjectFromSchemaRecord<typeof Canvas.info>;
 			type X7 = InsertableTreeFieldFromImplicitField<typeof Canvas.info.stuff>;
 		});
+
+		it("InsertableTreeNodeFromAllowedTypes with arrays of Unions", () => {
+			const sf = new SchemaFactory("test");
+
+			class Text1 extends sf.object("TextItem", { text: sf.string }) {}
+			class Text2 extends sf.object("TextItem2", { text: sf.string }) {}
+
+			type Text1Or2 = InsertableTreeNodeFromAllowedTypes<(typeof Text1 | typeof Text2)[]>;
+			// If the schema is unknown, we cannot guarantee that the input matches either schema,
+			// and only allow input valid in both, which is `never`:
+			allowUnused<requireAssignableTo<Text1Or2, never>>();
```

Comment:
Question: is this true of _any_ union, or just those where it is impossible to necessarily distinguish between them?

Reply:
Further reading answered my question.